pkg/build/api/v1/types.go

```diff
@@ -303,6 +303,9 @@ type BuildStrategy struct {
 
 	// CustomStrategy holds the parameters to the Custom build strategy
 	CustomStrategy *CustomBuildStrategy `json:"customStrategy,omitempty" description:"holds parameters to the Custom build strategy"`
+
+	// ExternalStrategy holds the parameters to the External build strategy
+	ExternalStrategy *ExternalBuildStrategy `json:"externalStrategy,omitempty" description:"holds parameters to the External build strategy"`
```

Comment:
I think, based on everything we've said, we want to make this be somewhat specific to workflow but not necessarily (implicitly) specific to Jenkins.  We want to pick the most general abstraction that solves the following problems:

1. Users want to define a complex workflow on their builds
2. A third party is going to implement that workflow
3. The third party needs to report back the workflow status to the status sub resource on the build config and clients need to read it (it's a versionable API)
4. A client can parameterize the external workflow (what parameters will we support?t)
5. If the workflow has a single definition (a Jenkinsfile), can we allow users to directly support that (are there any cases where a Jenkinsfile needs multiple files to work, because it includes another groovy script in the same dir?)